### PR TITLE
test: add module concatenation test case for ExternalsPlugin

### PR DESCRIPTION
**Pull request by aryanraj45**
**Summary**

This pull request aims to achieve 100% test coverage for [lib/ExternalsPlugin.js](cci:7://file:///Users/rajaryan/Projects/Web%20pack/webpack/lib/ExternalsPlugin.js:0:0-0:0) by introducing a new test case that specifically targets the `concatenatedModuleInfo` hook. 

Previously, the logic responsible for updating external module fragments during identifier renaming (due to naming collisions in a concatenated scope) was largely untested. This PR adds a comprehensive integration test in `test/configCases/externals/externals-concatenated-logic/` that:
1. Simulates a naming collision between a local variable and an external module's internal identifier.
2. Ensures that [ModuleConcatenationPlugin](cci:2://file:///Users/rajaryan/Projects/Web%20pack/webpack/lib/optimize/ModuleConcatenationPlugin.js:54:0-881:1) correctly renames the identifier and that [ExternalsPlugin](cci:2://file:///Users/rajaryan/Projects/Web%20pack/webpack/lib/ExternalsPlugin.js:19:0-90:1) updates the corresponding source fragments to reflect this change.
3. Ensures that [ExternalModule](cci:2://file:///Users/rajaryan/Projects/Web%20pack/webpack/lib/ExternalModule.js:640:0-1194:1) instances are correctly mark as "provided" in the test configuration to allow for successful concatenation.

**What kind of change does this PR introduce?**

`test`

**Did you add tests for your changes?**

Yes, added a new `configCases` test suite under `test/configCases/externals/externals-concatenated-logic/`.

**Does this PR introduce a breaking change?**

No.

**If relevant, what needs to be documented once your changes are merged or what have you already documented?**

N/A (Test-only change).

**Use of AI**
NO

### CODE REVIEW

File: test/configCases/externals/externals-concatenated-logic/test.config.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+"use strict";
+
+module.exports = {
+	timeout: 90000
+};
```
Comment (alexander-akait) [inline test/configCases/externals/externals-concatenated-logic/test.config.js]: Don't set `timeout`, it is only for edge cases, remove it

File: test/configCases/externals/externals-concatenated-logic/test.filter.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+"use strict";
+
+module.exports = () =>
+	!process.version.startsWith("v10.") && !process.version.startsWith("v12.");
```
Comment (alexander-akait) [inline test/configCases/externals/externals-concatenated-logic/test.filter.js]: rewrite it using `parserInt` and `majorVersion >= 14`

File: test/configCases/externals/externals-concatenated-logic/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+"use strict";
+
+const ExternalModule = require("../../../../lib/ExternalModule");
+
+/** @type {import("../../../../").Configuration} */
+module.exports = {
+	mode: "production",
+	target: "node14",
```
Comment (alexander-akait) [inline test/configCases/externals/externals-concatenated-logic/webpack.config.js]: Set `node`, no need to set to `14`

File: test/configCases/externals/externals-concatenated-logic/webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	optimization: {
+		concatenateModules: true,
+		minimize: false,
+		usedExports: true,
+		providedExports: true,
+		mangleExports: true
+	},
```
Comment (alexander-akait) [inline test/configCases/externals/externals-concatenated-logic/webpack.config.js:10]: `production` set everything to true, so  only `minimize` is needed here

**Comment by aryanraj45** [comment]
@alexander-akait done address all your review please take a look thanks !

File: test/configCases/externals/externals-concatenated-logic/webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+				compiler.hooks.compilation.tap("ExternalsTestPlugin", (compilation) => {
+					compilation.hooks.finishModules.tap(
+						"ExternalsTestPlugin",
+						(modules) => {
+							for (const m of modules) {
+								if (m instanceof ExternalModule && m.request === "path") {
+									const exportsInfo = compilation.moduleGraph.getExportsInfo(m);
+									exportsInfo.getExportInfo("join").provided = true;
+								}
+							}
```
Comment (alexander-akait) [inline test/configCases/externals/externals-concatenated-logic/webpack.config.js]: Do we real need it here? You can just add doNothing(join) to keep it in code

**Comment by aryanraj45** [comment]
@alexander-akait i used [doNothing({ join })]! However, we still need the `ExternalsTestPlugin` to bypass the concatenation bailout. Without marking exports as `provided`, `ModuleConcatenationPlugin` ignores external identifiers during renaming, which gave a  `SyntaxError: Identifier already declared`

**Comment by aryanraj45** [comment]
Pushed the code please have a look thanks!

File: test/configCases/externals/externals-concatenated-logic/webpack.config.js (hunk 5 above)
Comment (alexander-akait) [reply test/configCases/externals/externals-concatenated-logic/webpack.config.js]: I think we can remove it?

Comment (aryanraj45) [reply test/configCases/externals/externals-concatenated-logic/webpack.config.js]: are u asking about  ExternalsTestPlugin?

Comment (aryanraj45) [reply test/configCases/externals/externals-concatenated-logic/webpack.config.js]: it is mandatory because this plugin to maintain 100 %coverage for the hook concatenatedModuleInfo in externalplugin,js

**Comment by alexander-akait** [comment]
Let's fix tests

**Comment by aryanraj45** [comment]
@alexander-akait  updated the PR to Resolve the ci failure it was due to syntax error of redeclaring the variable now the externalplugin coverage is around 42%  (up from 0%). Please Take a look Thanks!

**Comment by aryanraj45** [comment]
Almost all the ci passes please check thanks!
